### PR TITLE
Remove unnecessary wrapper classes in ts.formatting.Rule

### PR DESCRIPTION
**Pull request by andy-ms**
We had a lot of classes whose sole purpose was to contain a few properties. It's simpler to just directly store these on the `Rule`.

### CODE REVIEW

File: src/services/formatting/rulesMap.ts [hunk 1]
```diff
@@ -1,60 +1,59 @@
...
-            return (row * this.mapRowLength) + column;
-        }
-
-        private FillRule(rule: Rule, rulesBucketConstructionStateList: RulesBucketConstructionState[]): void {
-            const specificRule = rule.descriptor.leftTokenRange.isSpecific() && rule.descriptor.rightTokenRange.isSpecific();
-
-            rule.descriptor.leftTokenRange.GetTokens().forEach((left) => {
-                rule.descriptor.rightTokenRange.GetTokens().forEach((right) => {
-                    const rulesBucketIndex = this.GetRuleBucketIndex(left, right);
+    export type RulesMap = (context: FormattingContext) => Rule | undefined;
```
Comment (mhegazy) [inline src/services/formatting/rulesMap.ts]: we need a different name for this type now.. RulesMap seems wrong.

Comment (mhegazy) [reply src/services/formatting/rulesMap.ts]: RuleMapper maybe..

Comment (andy-ms) [reply src/services/formatting/rulesMap.ts]: How about `GetRule`?

Comment (andy-ms) [reply src/services/formatting/rulesMap.ts]: @mhegazy Bump

Comment (mhegazy) [reply src/services/formatting/rulesMap.ts]: GetRule seems like a function to me and not a type name..

Comment (mhegazy) [reply src/services/formatting/rulesMap.ts]: @DanielRosenwasser or @sandersn might have better naming ideas.

File: src/services/formatting/rulesMap.ts [hunk 2]
```diff
@@ -1,60 +1,59 @@
-///<reference path='references.ts' />
+/// <reference path="rules.ts" />
```
Comment (mhegazy) [inline src/services/formatting/rulesMap.ts]: do not think we need any of these `/// <references` thourhg out the code base now that we have tsconfig.json files everywhere.. but i guess that is another change.

File: tests/baselines/reference/textChanges/extractMethodLike.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ===ORIGINAL===
 
-namespace M
```
Comment (mhegazy) [inline tests/baselines/reference/textChanges/extractMethodLike.js]: why did these change?

Comment (andy-ms) [reply tests/baselines/reference/textChanges/extractMethodLike.js]: I hit save in `unittests/textChanges.ts` and vscode automatically removed trailing whitespace from tests.
Moved that change to #19836.

Comment (mhegazy) [reply tests/baselines/reference/textChanges/extractMethodLike.js]: ah sorry.. i thought the code change somehow changed the baselines..

**Pull request merged**